pkg/dataobj/builder.go

```diff
@@ -193,11 +162,6 @@ func NewBuilder(cfg BuilderConfig, bucket objstore.Bucket, tenantID string) (*Bu
 // Once a Builder is full, call [Builder.Flush] to flush the buffered data,
 // then call Append again with the same entry.
 func (b *Builder) Append(stream logproto.Stream) error {
-	// Don't allow appending to a builder that has data to be flushed.
-	if b.state == builderStateFlush {
```

Comment:
I removed b.state completely, but I'm not sure if its the right decision. We still return ErrBufferFull when trying to add more data to a full object so I didn't think it was needed after also removing the Flush state.